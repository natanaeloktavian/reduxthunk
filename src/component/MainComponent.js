import React, { Component } from 'react'
import { TouchableHighlight, View, Text, StyleSheet,ScrollView } from 'react-native'

// @flow
function square(n: number): number {
  return n * n;
}

class MainComponent extends Component {
  render() {
    return (
      <ScrollView style={styles.container}>
        <Text style={styles.text}>Redux Examples {square("3")}</Text>
        <TouchableHighlight style={styles.button} onPress={() => this.props.fetchData()}>
          <Text style={styles.buttonText}>Load Data</Text>
        </TouchableHighlight>
        <View style={styles.mainContent}>
        {
            this.props.appData!=null && this.props.appData.isFetching && <Text>Loading</Text>
        }
        {
          (this.props.appData!=null && this.props.appData.data.length) ? (
            this.props.appData.data.map((person, i) => {
              return <View key={i} >
                <Text>Title: {person.title}</Text>
                <Text>Summary: {person.summary}</Text>
              </View>
            })
          ) : null
        }
        </View>
      </ScrollView>
    );
  }
}

styles = StyleSheet.create({
  container: {
    paddingTop: 10
  },
  text: {
    textAlign: 'center'
  },
  button: {
    height: 60,
    margin: 10,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#0b7eff'
  },
  buttonText: {
    color: 'white'
  },
  mainContent: {
    margin: 10,
  }
})

export default MainComponent
