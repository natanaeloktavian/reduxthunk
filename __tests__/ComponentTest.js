import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import 'react-native';
import React from 'react';
import MainComponent from '../src/component/MainComponent'

import renderer from 'react-test-renderer';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

it('renders correctly', () => {
  const tree = renderer.create(
    <MainComponent />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
