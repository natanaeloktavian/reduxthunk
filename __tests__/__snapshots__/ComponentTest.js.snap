// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "paddingTop": 10,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Redux Examples 
      9
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#0b7eff",
            "height": 60,
            "justifyContent": "center",
            "margin": 10,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        Load Data
      </Text>
    </View>
    <View
      style={
        Object {
          "margin": 10,
        }
      }
    />
  </View>
</RCTScrollView>
`;
